### ISSUE TITLE
[Deprecation] SharedArrayBuffer will require cross-origin isolation as of M91, around May 2021

### ISSUE THREAD

**Issue by AryanBeezadhur** [vercel/next.js#21708]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

14.15.0

**What browser are you using?**

Google Chrome Version 90.0.4400.8 (Official Build) dev (64-bit)

**What operating system are you using?**

Windows 10

**How are you deploying your application?**

`npm run dev` (`npx next dev`)

**Describe the Bug**

I get this warning in the console:

```
scheduler.development.js:298 [Deprecation] SharedArrayBuffer will require cross-origin isolation as of M91, around May 2021. See https://developer.chrome.com/blog/enabling-shared-array-buffer/ for more details.
```

![console warning](https://i.imgur.com/PxWOKBw.png)

**Expected Behavior**

No warnings in the console

**To Reproduce**

Run `npm run dev` or `npx next dev` in the terminal in a Next.js project.

**Comment by ploxx95** [comment]
i have same bug

**Comment by AryanBeezadhur** [comment]
*bump*

Do we have any information or an update on this?

**Comment by achrizal15** [comment]
saya juga mengalami hal yang sama

**Comment by agektmr** [comment]
I'm not entirely sure how next.js works, but since it has dependency with React, this issue might be related: https://github.com/facebook/react/issues/20829

**Comment by timneutkens** [comment]
Yep this is coming from React and has already been fixed (as linked above).

**Comment by gaearon** [comment]
17.0.2 is out with the fix. (But there was no real problem, it's a false positive.)

**Comment by iDVB** [comment]
@gaearon 
> 17.0.2 is out with the fix. (But there was no real problem, it's a false positive.)

We are still seeing this error with `17.0.2` for react and react-dom. We've also restarted the dev server.

**Comment by gaearon** [comment]
It does not contain that code so something’s wrong on your side. Check where the log is coming from and manually inspect how that gets pulled into your bundle.

**Comment by baileymckay** [comment]
I'm still getting this too.

react and react-dom are both on 17.0.2. I'm using next.js (11.0.1--latest), so it could be some build artifact. newSharedArrayBuffer appears twice in the next.js build file, though the one Chrome flags (I assume the first instance) seems to be coming from node_modules/util.

I know it's just a warning and it's nothing to worry about. I'm just tryin to figure out how it's still in the build. I cleared the cache, restarted the server, restarted my computer... The console warning is also appearing in my prod app.

<img width="384" alt="Screen Shot 2021-07-18 at 10 17 33 PM" src="https://user-images.githubusercontent.com/37872105/126093656-bde58158-2c3c-4924-aa7f-cb07ba6ee9fc.png">
<img width="604" alt="Screen Shot 2021-07-18 at 10 14 55 PM" src="https://user-images.githubusercontent.com/37872105/126093657-8c3026a7-aa27-4cad-9faa-bcee8dd66ad7.png">

**Comment by btahir** [comment]
> I'm still getting this too.
> 
> react and react-dom are both on 17.0.2. I'm using next.js (11.0.1--latest), so it could be some build artifact. newSharedArrayBuffer appears twice in the next.js build file, though the one Chrome flags (I assume the first instance) seems to be coming from node_modules/util.
> 
> I know it's just a warning and it's nothing to worry about. I'm just tryin to figure out how it's still in the build. I cleared the cache, restarted the server, restarted my computer... The console warning is also appearing in my prod app.
> 
> <img alt="Screen Shot 2021-07-18 at 10 17 33 PM" width="384" src="https://user-images.githubusercontent.com/37872105/126093656-bde58158-2c3c-4924-aa7f-cb07ba6ee9fc.png">
> 
> <img alt="Screen Shot 2021-07-18 at 10 14 55 PM" width="604" src="https://user-images.githubusercontent.com/37872105/126093657-8c3026a7-aa27-4cad-9faa-bcee8dd66ad7.png">

I'm still getting error as well (on React 17.02). Anybody found a workaround?

**Comment by agektmr** [comment]
Apparently `util` itself has solved the issue in [v0.12.4](https://github.com/browserify/node-util/releases/tag/v0.12.4), but multi-level dependencies are causing this issue. The followings are the libraries that depends on `util`:

* node-libs-browser@2.2.1
* assert@1.5.0
* assert@2.0.0

I've sent a pull request to fix the issue in `assert`.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
